main-settings/src/main/scala/sbt/std/InputWrapper.scala

```diff
@@ -228,6 +247,13 @@ sealed abstract class MacroPrevious[T] {
   def previous(implicit format: sjsonnew.JsonFormat[T]): Option[T] =
     macro InputWrapper.previousMacroImpl[T]
 }
+sealed abstract class MacroPreviousInThis[T] {
+  @compileTimeOnly(
+    "`previousInThis` can only be used within a task macro, such as :=, +=, ++=, or Def.task."
+  )
+  def previousInThis(implicit format: sjsonnew.JsonFormat[T]): Option[T] =
```

Comment:
This sounds too similar to `.previous` but `this` in sbt's context is not really clear.
`lastSeen` might be a better name?

Reply:
Yeah. I have thought about names involving last as well.

Reply:
I also considered `previousInThisTask` for the reason you mentioned.